network/wsNetwork.go

```diff
@@ -2350,18 +2348,35 @@ func (wn *WebsocketNetwork) updateMessagesOfInterestEnc() {
 	wn.messagesOfInterestEnc = MarshallMessageOfInterestMap(wn.messagesOfInterest)
 	wn.messagesOfInterestEncoded = true
 	atomic.AddUint32(&wn.messagesOfInterestGeneration, 1)
-	wn.messagesOfInterestCond.Broadcast()
+	select {
+	case wn.messagesOfInterestNotify <- struct{}{}:
+	default:
+		// it has been notified enough, it will pick up the latest when it actually runs
+	}
 }
 
 func (wn *WebsocketNetwork) postMessagesOfInterestThread() {
 	var peers []*wsPeer
-	wn.messagesOfInterestMu.Lock()
-	defer wn.messagesOfInterestMu.Unlock()
 	for {
-		wn.messagesOfInterestCond.Wait()
-		peers, _ = wn.peerSnapshot(peers)
-		for _, peer := range peers {
-			wn.maybeSendMessagesOfInterest(peer, wn.messagesOfInterestEnc)
+		select {
+		case <-wn.messagesOfInterestRefresh:
+			// if we're not a relay, and not participating, we don't need txn pool
+			wantTXGossip := wn.nodeInfo.IsParticipating()
+			if wantTXGossip && (wn.wantTXGossip != wantTXGossipYes) {
+				wn.RegisterMessageInterest(protocol.TxnTag)
+				atomic.StoreUint32(&wn.wantTXGossip, wantTXGossipYes)
+			} else if !wantTXGossip && (wn.wantTXGossip != wantTXGossipNo) {
+				wn.DeregisterMessageInterest(protocol.TxnTag)
+				atomic.StoreUint32(&wn.wantTXGossip, wantTXGossipNo)
+			}
```

Comment:
I thought there was an override so that non participating nodes would still get these messages. Looking back at the non-participating nodes PR I don't see that config. Is there another way that pending transactions are received?

Reply:
That's above in OnNetworkAdvance in `if wn.nodeInfo != nil && !wn.relayMessages && !wn.config.ForceFetchTransactions {` — nothing is sent to the messagesOfInterestRefresh channel if the config option is set.